src/ComplexTypes/BaseArrayOfType.php

```diff
@@ -15,26 +15,31 @@ abstract class BaseArrayOfType extends BaseType implements IteratorAggregate, Ar
      */
     const WRAPPED_PROPERTY = '';
 
+    #[\ReturnTypeWillChange]
     public function getIterator()
     {
         return new ArrayIterator($this->getWrappedProperty());
     }
 
+    #[\ReturnTypeWillChange]
     public function offsetExists($offset)
     {
         return isset($this->getWrappedProperty()[$offset]);
     }
 
+    #[\ReturnTypeWillChange]
     public function offsetGet($offset)
     {
         return $this->offsetExists($offset) ? $this->getWrappedProperty()[$offset] : null;
     }
 
+    #[\ReturnTypeWillChange]
     public function offsetSet($offset, $value)
     {
         throw new Exception('Writing to ArrayOf types using ArrayAccess not supported.');
     }
 
+    #[\ReturnTypeWillChange]
```

Comment:
We can just add the return types, instead of suppressing the deprecation warning, can't we? That would also make it forward-compatible with PHP 9.

Reply:
Guess we can indeed, because of the `"php": ">=7.0"` 🙂